core/domain/certificate_assessment_services.py

```
@@ -279,6 +297,345 @@ def _has_valid_distinct_assignment(
+    ).get()
+
+
+def _get_active_attempt_for_learner(
+    learner_id: str,
+) -> gae_models.CertificateAssessmentAttemptModel:
+    """Returns the learner's active assessment attempt or raises."""
+    attempt_model = _get_in_progress_attempt_for_learner(learner_id)
+    if attempt_model is None:
+        raise utils.ValidationError(
+            'No active certificate assessment attempt was found.'
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)